ODP-Runtime.md

```diff
@@ -124,28 +161,30 @@ Requirements we’d like to push upstream from a compatibility perspective:
 
 Best practices for ODP Platforms:
 
--   ODP Platforms SHOULD avoid using randomized ports when possible. For example, the nodemanager RPC port SHOULD NOT use the default ‘0’ (or random) value. Using randomized ports makes security extremely difficult as well as makes some parts of Apache Hadoop function incorrectly.
+-   ODP Platforms SHOULD avoid using randomized ports when possible. For example, the NodeManager RPC port SHOULD NOT use the default ‘0’ (or random) value. Using randomized ports may make firewall setup extremely difficult as well as makes some parts of Apache Hadoop function incorrectly.  Be aware that users MAY change these port numbers, including back to randomization.
 
 Compatibility
```

Comment:
How do applications obtain JAVA_HOME, HADOOP_CONF_DIR, HADOOP_YARN_HOME etc?

Do we assume that they are all set by the distro in the environment? In which environment, global or user? Or, will there be a prescribed way of locating the hadoop command and the run an env subcommand on it? 

Several distros today ensure that the hadoop command is in the path (sometimes via cascading symlinks). We have found that using this hadoop command is the best bet to bootstrap the app.